### PR TITLE
chromium: Fix build with clang 12

### PR DESCRIPTION
**Pull request by kraj**
Signed-off-by: Khem Raj <raj.khem@gmail.com>

### CODE REVIEW

**Review by rakuco — CHANGES_REQUESTED** [review]

File: recipes-browser/chromium/files/limits.patch [hunk 1]
```diff
@@ -0,0 +1,29 @@
+include <limits> for std::numeric_limits
+
+Fixes
+string_pool.h:256:34: error: no member named 'numeric_limits' in namespace 'std'
+    PERFETTO_DCHECK(value < std::numeric_limits<uint32_t>::max());
+                            ~~~~~^
+
+Upstream-Status: Pending
```
[the remark this answers is not in the document]
Comment (kraj) [reply recipes-browser/chromium/files/limits.patch]: done in v2

**Review by rakuco — APPROVED** [review]
lgtm!

**Comment by MastaG** [comment]
Works for me.

**Comment by shr-project** [comment]
```
ERROR: chromium-x11-89.0.4389.90-r0 do_patch: Fuzz detected:

Applying patch 0001-IWYU-include-limits-for-std-numeric_limits.patch
patching file third_party/perfetto/src/trace_processor/containers/string_pool.h
Hunk #1 succeeded at 23 with fuzz 1 (offset 1 line).


Applying patch 0001-IWYU-include-limits-for-std-numeric_limits-2.patch
patching file components/federated_learning/floc_constants.cc
Hunk #1 succeeded at 6 with fuzz 2 (offset 2 lines).


The context lines in the patches can be updated with devtool:

    devtool modify chromium-x11
    devtool finish --force-patch-refresh chromium-x11 <layer_path>

Don't forget to review changes done by devtool!

ERROR: chromium-x11-89.0.4389.90-r0 do_patch: QA Issue: Patch log indicates that patches do not apply cleanly. [patch-fuzz]
```

Needs small update for the new chromium version merged recently, will update locally.

**Comment by shr-project** [comment]
The first 2 patches are already included in 89.0.4389.90 currently in master, the 3rd patch does apply cleanly, but it might not be needed as I've already built this with clang12 I believe and the only failure I've seen was gold segfaulting in 1 build (from 3 I did in https://github.com/shr-project/test-oe-build-time)
```
clang-12: error: unable to execute command: Segmentation fault
clang-12: error: linker command failed due to signal (use -v to see invocation)

dmesg:
[225501.270737] ld.gold[1812127]: segfault at 7f120948a08a ip 000055797f63e0b7 sp 00007f15ff869c40 error 4 in x86_64-linux-gnu-ld.gold[55797f55d000+197000]
```

**Comment by kraj** [comment]
> The first 2 patches are already included in 89.0.4389.90 currently in master, the 3rd patch does apply cleanly, but it might not be needed as I've already built this with clang12 I believe and the only failure I've seen was gold segfaulting in 1 build (from 3 I did in https://github.com/shr-project/test-oe-build-time)
> 
> ```
> clang-12: error: unable to execute command: Segmentation fault
> clang-12: error: linker command failed due to signal (use -v to see invocation)
> 
> dmesg:
> [225501.270737] ld.gold[1812127]: segfault at 7f120948a08a ip 000055797f63e0b7 sp 00007f15ff869c40 error 4 in x86_64-linux-gnu-ld.gold[55797f55d000+197000]
> ```

third patch is needed when using gcc11 headers

**Comment by kraj** [comment]
closing this one

**Pull request closed without merging**